### ISSUE TITLE
Compiling on os Sierra - Compiler Error - typedef redefinition with different types ('int' vs 'enum')

### ISSUE THREAD

**Issue by wasam** [POV-Ray/povray#124]
OS: Sierra
Commit: 3cd6b6c

Compiling worked just fine, until I ran into the following error:
```
unix/syspovtimer.cpp:58:13: error: typedef redefinition with different types ('int' vs 'enum clockid_t')
typedef int clockid_t;
            ^
/usr/include/time.h:171:3: note: previous definition is here
} clockid_t;
```
so I took out Atom, the text editor of my choice, and out-commented the problematic line in `syspovtimer.cpp`, like so:
```cpp
#if !defined(HAVE_CLOCKID_T)
//typedef int clockid_t;
#endif
```
it then finished compiling without any further issues. 
I don't know if this is a good fix (I am quite unexperienced in C++).

**Comment by c-lipka** [comment]
I know it is not a good fix ;)
Just commenting out some code to work around incompatibilities with one particular platform tends to break things on other platforms.

Normally, the `./configure` build step should auto-detect whether your platform supports a `clockid_t` type, and set `HAVE_CLOCKID_T` accordingly, but for some reason that mechanism doesn't seem to be working properly on your platform.

Can you attach the `config.log` generated by the `./configure` build step?

**Comment by wasam** [comment]
[config.log.txt](https://github.com/POV-Ray/povray/files/537996/config.log.txt)

**Comment by c-lipka** [comment]
Thanks for the log; can you try replacing the following line in `unix/configure.ac`:

    AC_CHECK_TYPES([clockid_t])

with:

    AC_CHECK_TYPES([clockid_t], [], [], [[#include <time.h]])

and then rebuilding from scratch (including the prebuild step)?

**Comment by wasam** [comment]
I took the liberty and added an ">".
With that adjustment it compiles successfully on my other os Sierra machine.

**Comment by c-lipka** [comment]
Should be fixed with commit 8c8a18f.
